lib/shelf-reference.js

```diff
@@ -0,0 +1,113 @@
+         * @param {string} authToken
+         */
+        constructor(refName, authToken) {
+            this.refName = refName;
+            this.authToken = authToken;
+        }
+
+        /**
+         * Creates and returns a ShelfFile at the given path.
+         *
+         * @param {string} path
+         * @return {ShelfFile}
+         */
+        getArtifact(path) {
+            var uri;
+
+            logger.debug("In ShelfReference.getArtifact()");
+            uri = utils.buildUrl(path, host, this.refName);
+
+            return new ShelfFile(path, host, uri, this.authToken);
+        }
+
+        /**
+         * Sends and returns a request with the provided search criteria.
+         *
+         * @param {string} pathToSearch The path to perform the search on.
+         * @param {string} search The search criteria.
+         * @param {(string|string[])} sort A string or array of strings for
+         *                                 how the results will be sorted.
+         * @return {Promise}
+         */
+        search(pathToSearch, search, sort) {
+            var options, url;
+
+            logger.debug("In ShelfReference.search()");
+
+            url = utils.buildUrl(pathToSearch, host, this.refName);
+            options = this.getOptions();
+            options.url = `${url}/_search`;
+            options.body = {
+                search,
+                sort
+            };
+            options.json = true;
+
+            logger.debug(`POST request options: ${JSON.stringify(options, null, 4)}`);
+
+            return request.post(options).then((response) => {
+                var requestObj;
+
+                logger.debug("In request.get().then()");
+
+                requestObj = utils.handleResponse(response);
+
+                // Modify the links array to make ShelfFile objects from
+                // the links.
+                requestObj.links.forEach((val, index) => {
+                    var path, uri;
+
+                    path = val.replace(/.*artifact\//g, "");
+                    uri = utils.buildUrl(path, host, this.refName);
+                    requestObj.links[index] = new ShelfFile(path, host, uri, this.authToken);
+
+                    logger.debug(`Created new ShelfFile from link: ${JSON.stringify(requestObj.links[index], null, 4)}`);
+                });
+
+                return requestObj;
+            });
+        }
+
+        /**
+         * @typedef {Object} options
+         * @property {string} url
+         * @property {Object} headers
+         * @proptery {Boolean} resolveWithFullResponse
+         * @property {Boolean} simple
+         * @property {string} rejectUnathorized
+         */
+
+        /**
+         * Creates and returns the options for requests.
+         *
+         * @return {Object} options The options object used for requests.
+         */
+        getOptions() {
+            var options;
+
+            // TODO: Remove "rejectUnauthorized: false"
```

Comment:
Might I humbly suggest the swift and direct removal of the `rejectUnauthorized: false` from your options object?